src/wallet/coinselection.cpp

```diff
@@ -436,18 +441,25 @@ void SelectionResult::Clear()
 {
     m_selected_inputs.clear();
     m_waste.reset();
+    m_weight = 0;
 }
 
 void SelectionResult::AddInput(const OutputGroup& group)
 {
     util::insert(m_selected_inputs, group.m_outputs);
     m_use_effective = !group.m_subtract_fee_outputs;
+
+    m_weight += group.m_weight;
 }
 
 void SelectionResult::AddInputs(const std::set<COutput>& inputs, bool subtract_fee_outputs)
 {
     util::insert(m_selected_inputs, inputs);
     m_use_effective = !subtract_fee_outputs;
+
+    m_weight += std::accumulate(inputs.cbegin(), inputs.cend(), 0, [](int sum, const auto& coin) {
+        return sum + std::min(coin.input_bytes, 0) * WITNESS_SCALE_FACTOR;
```

Comment:
Should be `std::max(coin.input_bytes, 0)`.

Otherwise, this entire sum will be 0 most of the time (it's taking the smaller value between `input_bytes` and 0).

Reply:
Right thanks, I confused myself. Pushed.